### ISSUE TITLE
MacOS: Saving window position does not work (nightly, 1.15.0, metal and non-metal)

### ISSUE THREAD

**Issue by MathematicianOnGithub** [libretro/RetroArch#15701]
## Description

In MacOS Ventura 13.5 (22G74) window doesn't save position even when I toggle save also RetroArch starts at a weird initial position that partially hides the retroarch window.

### Expected behavior

In MacOS Retroarch should start in window mode in the center of the screen.
Retroarch should save window position if that option is toggled ON.

### Actual behavior

- Window starts in a weird position and not in the center of the screen.
- RetroArch does not save window position after starting Retroarch.

### Steps to reproduce the bug

1. Run Retroarch on MacOS Ventura 13.5 (22G74)

### Bisect Results

After installing Retroarch, run retroarch, weird initial start position even after restarting, even after using save window position feature, etc.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [version/commit]
nightly Fri 15, 2023, 1.15.0 (non-metal)
nightly Fri 15, 2023, 1.15.0 (metal)
1.15.0 non-metal
1.15.0 metal

### Environment information

- OS: MacOS Ventura 13.5 (22G74)

**Comment by MathematicianOnGithub** [comment]
I noticed in the code that it does save the positions to the retroarch.cfg.
However, it seems like the cocoa implementation never sets the initial position.

**Comment by warmenhoven** [comment]
Confirmed. I know I had this working at one point.

**Comment by warmenhoven** [comment]
I did work on this at one point, #14942 which fixed #14806. Not sure what broke it.

**Comment by personaone** [comment]
Yup - can confirm this, weird... I remember it was working at some point?

**Comment by warmenhoven** [comment]
Pretty sure https://github.com/libretro/RetroArch/pull/16436 fixes this.
